source/changelog.txt

```diff
@@ -12,6 +12,28 @@ Release Notes
    :depth: 1
    :class: singlecol
 
+v1.5.4
+------
+
+*Released July 31, 2022*
+
+Fixes a potential data corruption bug in
+``KeyVault.rewrapManyDataKey()`` when rotating encrypted data encryption
+keys backed by Azure or GCP key services.
+
+In previous versions of ``mongosh``, this bug occurrs when an
+Azure-backed or GCP-backed data encryption key being rewrapped required
+fetching an access token for decryption of the data encryption key.
+
+As a result of this bug, all data encryption keys being rewrapped are
```

Comment:
What does "material" mean here? Are the keys replaced by random cypher text or is the underlying data replaced by random cypher text? (You could potentially recover from the first problem. It would be much harder to recover from not the second issue.)

Reply:
@addaleax are you able to answer this question, and also if we should adjust the text to be clearer about what's meant by "material"?

Reply:
> What does "material" mean here?

It refers to the plaintext of the data encryption key, something that the user or the database never interacts with directly.

libmongocrypt creates data keys by generating random data (= key material), then encrypting it using the KMS and storing it in the key vault. When user data is encrypted/decrypted, it fetches the data key from the key vault, decrypts its contents using the KMS to get the key material, then encrypts/decrypts the data using that key material. With this bug, the key material is replaced with random data, leaving no way to decrypt user data.

> also if we should adjust the text to be clearer about what's meant by "material"?

These release notes were user for driver and libmongocrypt releases, and the wording here has been signed off on by product/drivers leads. I’d not worry about the exact wording here in the docs too much.

Reply:
Thanks for the clarification Anna. I'll keep the wording here as-is.

Reply:
Understood, I just wish they would have run the text past one of us first.